tests/system/feature_store/test_spark_engine.py

```diff
@@ -558,3 +558,54 @@ def test_mix_of_partitioned_and_nonpartitioned_targets(self):
             check_index_type=True,
             check_column_type=True,
         )
+
+    def test_write_empty_dataframe_overwrite_false(self):
+        name = "test_write_empty_dataframe_overwrite_false"
+
+        path = "v3io:///bigdata/test_write_empty_dataframe_overwrite_false.parquet"
+        fsys = fsspec.filesystem(v3iofs.fs.V3ioFS.protocol)
+        empty_df = pd.DataFrame(
+            {
+                "time": [
+                    pd.Timestamp("2021-01-10 10:00:00"),
+                ],
+                "first_name": ["moshe"],
+                "data": [2000],
+            }
+        )[0:0]
+        empty_df.to_parquet(path=path, filesystem=fsys)
+
+        source = ParquetSource(
+            "myparquet",
+            path=path,
+            time_field="time",
+        )
+
+        feature_set = fs.FeatureSet(
+            name=name,
+            entities=[fs.Entity("first_name")],
+            timestamp_key="time",
+            engine="spark",
+        )
+
+        output_path = "v3io:///bigdata/test_write_empty_dataframe_overwrite_false/"
+        targets = [
+            ParquetTarget(
+                name="pq",
+                path=output_path,
+                partitioned=False,
+            ),
+        ]
+
+        fs.ingest(
+            feature_set,
+            source,
+            run_config=fs.RunConfig(local=False),
+            targets=targets,
+            overwrite=False,
+            spark_context=self.spark_service,
+        )
+
+        # check that no files were written
+        with pytest.raises(FileNotFoundError):
+            pd.read_parquet(output_path)
```

Comment:
Is this expected to ever work? Since we're now always adding `run_id` in the path, wouldn't this fail also if you do ingest real data and a file is created? I think the correct usage should be to extract the output path from the feature-set's target.